main.tf

```diff
@@ -1,15 +1,17 @@
 resource "aws_iam_policy" "main" {
   name        = var.policy_name
+  name_prefix = var.name_prefix
   path        = var.path
   description = var.description
   policy      = var.policy
-  tags = {
-    name        = var.policy_name
-    environment = var.environment
-  }
+  tags = merge({
+    name = var.policy_name
+    },
+  var.tags)
 }
 
 resource "aws_iam_policy_attachment" "main" {
+  count      = local.create_policy_attachment ? 1 : 0
```

Comment:
With all of the above comments, this can then be modified to
 ` count      = local.create_policy_attachment && var.policy_attachment_name !=null ? 1 : 0